File: LeetCode/1207/solution.java
// 1207. Unique Number of Occurrences
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> mp = new HashMap<>();
        int[] freq = new int[1001];
        for(int a: arr){
            int val = mp.getOrDefault(a, 0);
            freq[val]--;
            mp.put(a, val+1);
            freq[val+1]++;
        }
        for(int a: freq){
            if(a>1) return false;
        }
        return true;
    }
}

File: LeetCode/1207/solution.cpp
// 1207. Unique Number of Occurrences
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> mp;
        int freq[1001] = {0};
        for(auto a: arr){
            freq[mp[a]]--;
            mp[a]++;
            freq[mp[a]]++;
        }
        for(auto a: freq){
            if(a>1) return false;
        }
        return true;
    }
};